src/adera/Machines/Rocket.cpp

```diff
@@ -137,37 +108,29 @@ void SysMachineRocket::update_physics()
 
         if (WireData *throttle = machine.m_wiThrottle.connected_value())
         {
-            Percent *percent = std::get_if<Percent>(throttle);
-
-            float thrust = 10.0f; // temporary
-
-            //std::cout << percent->m_value << "\n";
-
-            Vector3 thrustVec = compTf->m_transform.backward()
-                                    * (percent->m_value * thrust);
-
-            SysPhysics_t::body_apply_force(*compRb, thrustVec);
+            Percent *pPercent = std::get_if<Percent>(throttle);
```

Comment:
Ah, ok, so 

```
// Supported data types
using WireData = std::variant<wiretype::Attitude,
                              wiretype::AttitudeControl,
                              wiretype::Percent,
                              wiretype::Deploy>;
```

Can't say I'm a huge fan of using std::variant here.

IMHO the wires should either be completely templated on the payload all the way through, so that the compiler is enforcing that the wire can't be connected to the wrong kind of thing. For example, you're gonna have a hard time plugging an ethernet cable into an HDMI port.

Or, if not templated throughput, then it should be completely type-erased, as in, void* + size_t, so that incorrectly connected wires get the correct "garbage in = garbage out" behavior that you would expect. At the receiving end of the wire data, we would need to do a size check to make sure we aren't evaluating data that is sized incorrectly, but beyond that, just do the "wrong thing".

Reply:
Paging @Capital-Asterisk, I didn't design the system so I'm not really in a place to reason about the long-term consequences of doing those things.

Reply:
Yep, the wiring system definitely needs some beefing up. The entire thing needs to be nuked and rebuilt. I've written an issue on it, but I haven't submitted it yet.